elastic/integrations
# [azure logs] add routing integration to use only one azure-eventhub input

## Proposed commit message




Switch the integration package from the one-input-per-data-stream model to the one-input model.

One input per data stream model:

![image](https://github.com/user-attachments/assets/ce60cd18-80fc-4805-aebf-322dcfd5f374)

One input model:

![image](https://github.com/user-attachments/assets/b8ae0dbd-6ca6-4caa-ad5c-3c4a6f573dee)


In the one-input model, there is only one azure-eventhub input running and sending events to the `events` data stream. In the `events` data stream, the ingest pipeline performs these tasks:

- discover and set the `event.dataset` field using the `category` field in the event.
- use the `event.dataset` field to reroute the event to the target data stream.

The discover process uses the following logic:

- if the event doesn't have a category, it sets `event.dataset` to `azure.eventhub` (the generic integration)
- if the event does have a category, it sets `event.dataset` to `azure.platformlogs` (it's probably an Azure log)
- if the event category is supported, it sets `event.dataset` to specific one like `azure.activitylogs` or `azure.signinlogs`.

After the discovery step, the routing rules use the `event.dataset` value to forward the events to the best available target data stream.


## Checklist


## Author's Checklist



## How to test this PR locally



- Bump the integration version
- Build and start a local stack
- Install the Azure Logs package (turn on v2 and turn off all v1s)
- Send test documents and check they are redirected to the data stream

### Bump the integration version

I'll update the version and changelog later to avoid conflicts. In the meantime, a simple bump would do:

```diff
diff --git a/packages/azure/changelog.yml b/packages/azure/changelog.yml
index bd20d10043..6da52b0424 100644
--- a/packages/azure/changelog.yml
+++ b/packages/azure/changelog.yml
@@ -1,3 +1,8 @@
+- version: "1.20.0+build18"
+  changes:
+    - description: Add Azure Logs integration v2 (preview)
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/11432
 - version: "1.19.4"
   changes:
     - description: Fix destination.geo.region_name mapping.
diff --git a/packages/azure/manifest.yml b/packages/azure/manifest.yml
index be7dccf294..895d9a8b6e 100644
--- a/packages/azure/manifest.yml
+++ b/packages/azure/manifest.yml
@@ -1,6 +1,6 @@
 name: azure
 title: Azure Logs
-version: 1.19.4
+version: "1.20.0+build18"
 description: This Elastic integration collects logs from Azure
 type: integration
 icons:
```

### Build and start a local stack

```sh
elastic-package build && elastic-package stack up -d -v --version 8.16.1
```

### Install the Azure Logs package

- Turn on the v2 preview integration
- Turn off all the v1 integrations

### Send test documents and check they are redirected to the data stream

I use the [eventhubs](https://github.com/zmoog/eventhubs) CLI tool to send JSON documents to an event hub.

Set up some environment variables:

```sh
export EVENTHUB_CONNECTION_STRING="[redacted]"
export EVENTHUB_NAMESPACE="[redacted]"
export EVENTHUB_NAME="logs"
```

We can then `cat` the test document and pipe them into an event hub using `eh -v eventdata send-batch`:

```sh
cat data_stream/provisioning/_dev/test/pipeline/test-provisioninglogs-raw.log | eh -v eventdata send-batch 
```

All supported documents should land into the expected data stream. Any unsupported document should land into the platform logs data stream.

## Related issues


- Closes https://github.com/elastic/integrations/issues/11251

## Screenshots



### Integration setup

![CleanShot 2024-12-03 at 15 46 55@2x](https://github.com/user-attachments/assets/bca3a5b8-7d57-4ba3-88fc-6ce210271a68)

### Sample from the integration docs

![CleanShot 2024-12-03 at 23 25 35@2x](https://github.com/user-attachments/assets/b14080ae-37c5-4db0-a788-578d9ef183bc)
